cps/src/main/js/workflow-editor.js

```diff
@@ -12,98 +9,115 @@ $('.textarea-handle', wrapper).remove();
 jenkinsJSModules.import('ace-editor:ace-editor-122')
     .onFulfilled(function (acePack) {
         
-        // The 'ace-editor:ace-editor-122' plugin supplies an "ACEPack" object.
-        // ACEPack understands the hardwired async nature of the ACE impl and so
-        // provides some async ACE script loading functions.
+        $('.workflow-editor-wrapper').each(function() {
+            initEditor($(this));        
+        });
         
-        acePack.edit('workflow-editor', function() {
-            var ace = acePack.ace;
-            var editor = this.editor;
+        function initEditor(wrapper) {
+            var textarea = $('textarea', wrapper);
+            var aceContainer = $('.editor', wrapper);
             
-            // Attach the ACE editor instance to the element. Useful for testing.
-            var $wfEditor = $('#workflow-editor');
-            $wfEditor.get(0).aceEditor = editor;
-
-            acePack.addPackOverride('snippets/groovy.js', '../workflow-cps/snippets/workflow.js');
-
-            acePack.addScript('ext-language_tools.js', function() {
-                ace.require("ace/ext/language_tools");
+            $('.textarea-handle', wrapper).remove();
+            
+            // The ACE Editor js expects the container element to have an id.
+            // We generate one and add it.
+            editorIdCounter++;
+            var editorId = 'workflow-editor-' + editorIdCounter;
+            aceContainer.attr('id', editorId);
+            
+            // The 'ace-editor:ace-editor-122' plugin supplies an "ACEPack" object.
+            // ACEPack understands the hardwired async nature of the ACE impl and so
+            // provides some async ACE script loading functions.
+            
+            acePack.edit(editorId, function() {
+                var ace = acePack.ace;
+                var editor = this.editor;
                 
-                editor.$blockScrolling = Infinity;
```

Comment:
[Ignore whitespace](https://github.com/jenkinsci/workflow-plugin/pull/334/files?w=1#diff-63ad0d7784893a9dca6c64f8c55aa693) to review.